extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -141,6 +143,9 @@ public void build(CombinedIndexBuildItem indexBuildItem, BuildProducer<Reflectiv
                     new ReflectiveClassBuildItem(false, false, ObjectMapperSerializer.class, ObjectMapperDeserializer.class));
             collectSubclasses(toRegister, indexBuildItem, ObjectMapperSerializer.class);
             collectSubclasses(toRegister, indexBuildItem, ObjectMapperDeserializer.class);
+
+            // Make the `io.quarkus.jackson.runtime.ObjectMapperProducer` bean cannot be removed.
```

Comment:
The comment is off but I merged it anyway as I prefer having this in 1.11.2.Final.